envoy/server.go

```diff
@@ -0,0 +1,437 @@
+package envoy
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/Nitro/sidecar/catalog"
+	"github.com/Nitro/sidecar/config"
+	"github.com/Nitro/sidecar/service"
+	api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	endpoint "github.com/envoyproxy/go-control-plane/envoy/api/v2/endpoint"
+	listener "github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	route "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	hcm "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	tcpp "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/tcp_proxy/v2"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
```

Comment:
Let's not call this `discovery` because we already have our own. Let's call it `envoy_discovery` or similar.